x-pack/legacy/plugins/infra/public/components/logging/log_text_stream/scrollable_log_text_stream_view.tsx

```diff
@@ -102,6 +102,11 @@ export class ScrollableLogTextStreamView extends React.PureComponent<
         targetId: null,
         items: [],
       };
+    } else if (hasItems && nextItems.length > prevState.items.length) {
```

Comment:
Is looking at the number of items a sufficient criterion? What if the list of items is replaced by a list of the same length?

Reply:
The way the app is built right now, this never happens. The only time the list of items is wholly replaced is when the user jumps to a timestamp that requires all logs data to reload, which unmounts this component and replaces it with the loading screen.

However this also seems to work if I change it to `(nextItems.length !== prevState.items.length || nextItems[0] !== prevState.items[0])` so that could make this fix more future-proof.

Reply:
True, that shouldn't happen in the context of the app. I like your improvement, though :+1: Let's hope we find an opportunity to clean up the whole state and rendering code soon (and move to hooks and `react-virtualized` :innocent:) :crossed_fingers: I hope #49154 will give us the opportunity.